src/Storages/ObjectStorage/DataLakes/DeltaLakeMetadataDeltaKernel.h

```diff
@@ -22,12 +22,11 @@ namespace DB
 class DeltaLakeMetadataDeltaKernel final : public IDataLakeMetadata
 {
 public:
-    using ConfigurationObserverPtr = StorageObjectStorage::ConfigurationObserverPtr;
     static constexpr auto name = "DeltaLake";
 
     DeltaLakeMetadataDeltaKernel(
         ObjectStoragePtr object_storage_,
-        ConfigurationObserverPtr configuration_);
+        StorageObjectStorageConfigurationObserverPtr configuration_);
```

Comment:
Such a long name 😢

Reply:
May be raname `Observer` -> `Weak` as it is shorter and also consistent with `DB::Context`, where we have `ContextPtr` and `ContextWeakPtr`?